src/xstate/actors/audioPlayer/audioPlayerMachine.ts

```diff
@@ -0,0 +1,1027 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable import/prefer-default-export */
+/* eslint-disable no-restricted-syntax */
+/* eslint-disable jsdoc/check-tag-names */
+/* eslint-disable max-lines */
+import random from 'lodash/random';
+import { assign, createMachine, send, spawn } from 'xstate';
+import { pure, stop } from 'xstate/lib/actions';
+
+import { createRadioMachine } from '../radio/radioMachine';
+import { createRepeatMachine } from '../repeatMachine/repeatMachine';
+
+import AudioPlayerContext from './types/AudioPlayerContext';
+import AudioPlayerEventType from './types/AudioPlayerEventType';
+
+import { fetcher, getAvailableReciters } from 'src/api';
+import AudioData from 'types/AudioData';
+import Reciter from 'types/Reciter';
+import VerseTiming from 'types/VerseTiming';
+import Word from 'types/Word';
+
+/**
+ * check if currentTime is within range timestampFrom and timestampTo
+ *
+ * example:
+ * - timestampFrom = 10, timestampTo = 20, currentTime = 10 should return true
+ * - timestampFrom = 10, timestampTo = 20, currentTime = 11 should return true
+ * - timestampFrom = 10, timestampTo = 20, currentTime = 20 should return false
+ *
+ * @param {number} currentTime
+ * @param {number} timestampFrom
+ * @param {number} timestampTo
+ * @returns {boolean} isWithinRange
+ */
+const isCurrentTimeInRange = (currentTime: number, timestampFrom: number, timestampTo: number) =>
```

Comment:
I suggest this being moved to a helper file since the file is very bloated already and hard to follow.